lib/mix/tasks/phoenix.gen.html.ex

```diff
@@ -66,7 +66,8 @@ defmodule Mix.Tasks.Phoenix.Gen.Html do
     end
   end
 
-  defp validate_args!([_, plural | _] = args) do
+  defp validate_args!([singular, plural | _] = args) do
+    Mix.Phoenix.check_module_name_availability!(singular)
```

Comment:
It doesn't work even for the model generator because we need to consider the namespace. If we don't consider the namespace, we won't even detect the case in #921 as the conflict there is effectively with a namespace module.